controllers/dscinitialization/dscinitialization_controller.go

```diff
@@ -435,3 +444,35 @@ func (r *DSCInitializationReconciler) watchDSCResource(ctx context.Context) []re
 	}
 	return nil
 }
+
+func (r *DSCInitializationReconciler) createAuthSingleton(ctx context.Context) error {
+	authList := &serviceApi.AuthList{}
+	err := r.Client.List(ctx, authList)
+	if err != nil {
+		return err
+	}
+
+	if len(authList.Items) == 0 {
```

Comment:
nit: you can eventually replace this list+create with a single create call and ignore the error in case it reports "already exists"

Reply:
:+1: